pkg/services/ngalert/notifier/channels/notifiers.go

```diff
@@ -0,0 +1,31 @@
+package channels
+
+import (
+	"sort"
+
+	"github.com/grafana/grafana/pkg/services/alerting"
+)
+
+// notifiersMetadata stores all the active notification channels.
+// The Factory field in the struct is not meant to be used and not required to be set.
+var notifiersMetadata = make(map[string]*alerting.NotifierPlugin)
```

Comment:
Can we avoid using a global variable. We are specifically moving away from global state in the Grafana backend.

Reply:
Since this is going to be a static config with almost no changes required (unless when notification is modified/added/deleted), I am returning from the Alertmanager instead of going through the dependency injection

Reply:
(i.e. removed the global variable now :))